src/query/ee/src/storages/fuse/io/snapshots.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Databend Cloud
+//
+// Licensed under the Elastic License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.elastic.co/licensing/elastic-license
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Read all the referenced segments by all the snapshot file.
+// limit: limits the number of snapshot files listed
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use common_exception::Result;
+use common_storages_fuse::io::SnapshotLiteExtended;
+use common_storages_fuse::io::SnapshotsIO;
+use storages_common_table_meta::meta::Location;
+use tracing::info;
+
+#[allow(clippy::too_many_arguments)]
+#[async_backtrace::framed]
+pub async fn get_snapshot_referenced_segments<T>(
```

Comment:
these two functions is not very business specific, maybe it's also feasible to be placed in the apache licensed part?

Reply:
We can't place ee functions(only used by the ee) follow apache license.